Foundation/NSHost.swift

```diff
@@ -33,7 +44,21 @@ open class Host: NSObject {
         _type = type
     }
     
-    static internal let _current = Host(nil, .current)
+    static func currentHostName() -> String {
+        let hname = UnsafeMutablePointer<Int8>.allocate(capacity: Int(NI_MAXHOST))
+        defer {
+            hname.deinitialize()
+            hname.deallocate(capacity: Int(NI_MAXHOST))
+        }
+        let r = gethostname(hname, Int(NI_MAXHOST))
+        if r < 0 || hname[0] == 0 {
+            print("gethostname returned empty string. Will use localhost")
```

Comment:
the log here is not needed; either fatalError or return a fallback (which I would claim the fallback is reasonable)